src/rgw/rgw_rest_s3.cc

```diff
@@ -6176,7 +6213,7 @@ void aws_response_handler::init_response()
 void aws_response_handler::init_success_response()
 {
   m_buff_header.clear();
-  header_size = create_header_records();
+  header_size = create_header_records() + create_header_continuation() + create_header_progress() + create_header_stats();
```

Comment:
no sure it's correct according to [https://docs.amazonaws.cn/en_us/AmazonS3/latest/API/RESTSelectObjectAppendix.html](spec)
each type should stand by itself (not to concatenate headers)

please run a simple query test(success and error)

Reply:
Got it. Thanks @galsalomon66  Each type will have separate headers and payload.